### ISSUE TITLE
Problem with spawners

### ISSUE THREAD

**Issue by baticag** [tastybento/askyblock#453]
Hi Tastybento,

My players started experiencing one big problem about a year ago, but I haven't known and I still don't if that is Minecraft issue or it's a bug related to ASkyBlock plugin, that's why I am asking for help here.

Server has custom island schematics with mob spawners on nether island. Spawners can be changed by buying a mob egg from shop. Problems happen when players want to change spawners with egg and it isn't happening for all players.

When they try to change spawner type with egg, one of this can happen:
1) They change it successfully, they see the new mob in spawner and everything is okay
2) They change it successfully and they see the new mob, but after couple of seconds it changes back to old mob and they wasted the egg they bought.
3) They change it successfully and they see the new mob, but when leave nether or leave server (or I guess when the chunk unloads)  it changes back to old mob type.

I have tried debugging this one for months and tried reproducing it and recording with players, but caught nothing. Now it is really getting on my players and mine nerves...

Can you please help me to try and sort this out?
Thanks in advance!

**Comment by tastybento** [comment]
It doesn't sound like anything to do with ASkyBlock, but I'll try and help. What plugin are you using to enable the mob spawner switching using the eggs? That's the first place to start.

**Comment by baticag** [comment]
Thank you very much I really appreciate it!

We don't use any plugins that have anything to do with spawners that's the trick. Vanilla minecraft supports mob spawner changing with mob eggs since 1.7.9. Some eggs are given with Essentials /give command and some from ASB challenges. Today one player got wolf egg from challenge and tried, but ended as 2. from above.

**Comment by tastybento** [comment]
Oh wow, I can't believe I'm still learning things about this game. Okay, I'll have a look. Question though - does this happen only in the ASkyBlock world? Also, what server version are you using?

**Comment by baticag** [comment]
Server is running latest Spigot 1.11 (updated every morning) and latest ASB version. Just to note that players have spawners only in ASB nether world and that's the world I am talking about.

**Comment by tastybento** [comment]
I tested it with 1.11.2 and default nether island and everything seems to work okay. Of course, the right block needs to be near the spawner for the creature to spawn, e.g., grass. I tried a few things like stopping the server and restarting but I couldn't cause it to happen.

I'm not sure what to do. It almost sounds like it's something to do with the server not saving the chunks properly or a client<>server lag issue. 

Have you seen this happen yourself? It sounds like it's intermittent. It may require some debug plugin to track when a player uses an egg on a spawner and log it along with what the state is.
